### ISSUE TITLE
Metrics for missing timestep of track_to_predict

### ISSUE THREAD

**Issue by jiaxiaosong1002** [waymo-research/waymo-open-dataset#374]
Hi, after examining the data, I have noticed that in the training/validation set of Waymo Open Motion Dataset, even for those agents with track_to_predict as 1 (i.e, suggested target agents), their future states are not guaranteed to be fully observable. In other words, the future 80 frames of the target agents are not all valid.

So, I am wondering, how the leaderboard compute metrics in these situations? For example, how to compute ADE when there is one missing future timestep? How to compute FDE/MR/mAP when the timestep at 30/50/80 are missing? 
From https://github.com/waymo-research/waymo-open-dataset/issues/302, it seems that the ground-truth coordinates of the missing timesteps are setting as (-1, -1)? If this is the case, it seems like a little bit unreasonable groud-truth to learn.

 Why not to only let those agents with fully observable data (1s + 8s) be possible to have track_to_predict as 1? In this way, the data are cleaner for learning-based methods? I would like to learn about the concerns in the engineering and practical application perspectives.

Thank you for your time and help!

**Comment by scott-ettinger** [comment]
Thanks for asking this question.  Each future state has an associated valid flag.  These can be used to mask out computations for losses such that only the valid steps get propagated for learning.  

This was a design decision.  Choosing only objects with fully valid future data would eliminate too many of the interesting objects in the dataset as occlusions are very common.  The chosen objects in tracks_to_predict for the standard set do have a minimum ratio of valid states (I believe it is 65%).  The interactive set has a much higher ratio of valid states with more fully valid trajectories as the segments were chosen around these objects.

As for the metrics, invalid states are ignored in the ADE computation (they are not included in the average).  If the object does not exist at the final time step, then the object is not included in the FDE, MR, or mAP computation. 

Please let me know if you have further questions.

**Comment by jiaxiaosong1002** [comment]
Thanks a lot for your detailed explanation which is very helpful and totally solves my concerns. They are indeed reasonable ways of processing data. Thanks again for your kind help!
